lib/mongoose_push/application.ex

```diff
@@ -86,6 +88,39 @@ defmodule MongoosePush.Application do
     end)
   end
 
+  defp metrics do
+    [
+      # Summary is not yet supported in TelemetryMetricsPrometheus
+      Telemetry.Metrics.counter("mongoose_push.push.time",
+        tags: [:status, :service, :type, :reason],
+        tag_values: fn metadata ->
+          case metadata.status do
+            :success ->
+              Map.merge(
+                %{
+                  type: :success,
+                  reason: :success
+                },
+                metadata
+              )
+
+            :error ->
+              Map.merge(
+                %{type: :generic},
+                metadata
+              )
+          end
+        end
+      ),
+
+      # measurement is ignored in Counter metric
+      Telemetry.Metrics.counter("mongoose_push.supervisor.init.count", tags: [:service]),
+      Telemetry.Metrics.counter("mongoose_push.apns.state.init.count"),
+      Telemetry.Metrics.counter("mongoose_push.apns.state.terminate.count", tags: [:reason]),
```

Comment:
```suggestion
      Telemetry.Metrics.counter("mongoose_push.apns.state.terminate.count", tags: [:error_reason]),
```

Reply:
If this is easier we can change the telemetry events to also use this naming, but as mentioned above:
```In our code I guess the old name is fine without the error_ prefix, but this will be external name, for someone that doesn't see the code. We need the context.```

Reply:
Likewise, can you please explain? I'm just more and more lost 😕